src/components/dock/dock-button/dock-button.scss

```diff
@@ -62,8 +69,27 @@ limel-popover {
 }
 
 .icon {
+    position: relative;
     flex-shrink: 0;
     width: calc(var(--dock-item-height) - 1rem);
     height: calc(var(--dock-item-height) - 1rem);
     color: var(--dock-item-icon-color, var(--limel-dock-item-text-color));
+
+    &:before {
+        pointer-events: none;
+        position: absolute;
+        inset: 0;
+        background-color: var(
+            --dock-background-color,
+            rgb(var(--contrast-100))
+        );
+        // 👇 Below is a hidden and private prop
+        // only for usage in Lime CRM components.
+        // It enables our customers to override
+        // the default Home icon, using an external URL
+        // or an SVG code. Since it defaults to `undefined`
+        // no pseudo element will be rendered, unless
+        // the consumer defines something.
```

Comment:
I would rather say that this is a perfectly reasonable feature to have in `limel-dock`, and I'd be happy to approve it (once reviewed, obviously); it's whether we should enable the feature in _Lime CRM_ that we need to discuss with the team and with Product, before we enable anything.

With that said, I love that you're doing this as a hackday thing! 👍 ❤️

Reply:
Thanks Adrian. I hope the demo explained well-enough how it works.
I'm however thinking to improve this code a little bit. Since we currently have a problem in this solution, and that is: if we have an image file (`.svg`, `.png`) with a large canvas size, it won't fit in well in the UI. Fixing it is however very easy. Needs a little refactoring here and a bit of change in the Lime Admin plugin.

Reply:
however, that might also be a good limitation. This way, we can ensure that they don't carelessly load a huge PNG file in there that slows down the loading speed of the client... 

On the other hand, the target audience is consultants, and hopefully they are are of these technical dos and donts. So easing their lives must be a higher priority for us. If I fix it, we don't need to give them lots of guidelines and how to documents.

Reply:
I did the changes I wanted and it's way better. Now @Befkadu1 and I should just tweak a few things in the Lime Admin plugin.

Reply:
I still don't understand why we should hide this feature in Lime Elements? Why don't we consider it a proper feature and document it, instead of consider it a hack and hiding it? 🤨

Reply:
Ah, ok, is it because supplying a url and loading the image from there, rather than supplying the actual image, is pretty hacky? Ok, I think I've changed my mind already. Keep it hidden. Let's add this feature properly at some point instead 👍

Reply:
(I have this vision that we'd implement an `Icon` interface, with `name`, `color`, `backgroundColor`, etc, and then we could add `src` or something to that, and the consumer could supply their own svg data there. Then, if that's set, limel-icon would render that data, instead of getting an icon using the `name` property.